modules/andes-core/broker/src/main/java/org/wso2/andes/store/cassandra/CQLBasedMessageStoreImpl.java

```diff
@@ -549,11 +507,11 @@ public void moveMetaDataToQueue(long messageId,
             throw new AndesException("Message MetaData not found to move the message to Dead Letter Channel");
         }
 
-        List<AndesRemovableMetadata> removableMetadataList = new ArrayList<AndesRemovableMetadata>(1);
-        removableMetadataList.add(new AndesRemovableMetadata(messageId, currentQueueName, currentQueueName));
+        List<Long> removableMetadataList = new ArrayList<Long>(1);
+        removableMetadataList.add(messageId);
 
         addMetaDataToQueue(targetQueueName, metadataList.get(0));
```

Comment:
It will only be empty if the method 'getNextNMessageMetadataFromQueue' returns an empty list. An Andes exception is thrown in that case. Therefore, will not proceed to throw a NPE.